app/models/message.rb

```diff
@@ -5,6 +5,8 @@ class Message < ApplicationRecord
   validates :text, presence: true
   validate :user_belongs_to_chat, on: :create
 
+  scope :unread_messages, ->(user) { where(seen: false).where.not(user_id: user.id) }
```

Comment:
I would name this `unread_for`  ...  so that you then call it like `messages.unread_for(current_user)` or something similar that is more easy to read.